boa3/compiler/codegenerator/codegenerator.py

```diff
@@ -1011,49 +1026,191 @@ def convert_get_sub_array(self, value_addresses: List[int] = None):
                     self.fix_negative_index(code)
                 self.fix_negative_index()  # fix the last value sign
 
+                if negative_stride:
+                    """
+                    If stride is negative, then the array was reversed, thus, lower and upper should be changed 
+                    accordingly.
+                    array[lower:upper:-1] == reversed_array[len(array)-lower-1:len(array)-upper-1]
+                    """
+                    # calculates corresponding upper
+                    # upper = len(array)-upper-1
+                    self.duplicate_stack_item(3)
+                    self.convert_builtin_method_call(Builtin.Len)
+                    self.swap_reverse_stack_items(2)
+                    self.convert_operation(BinaryOp.Sub)
+                    self.__insert1(OpcodeInfo.DEC)
```

Comment:
this block of code seems to be used in many places, it could be implemented as a method to reuse the code